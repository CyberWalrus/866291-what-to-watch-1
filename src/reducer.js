import films from "./mock/films.js";
const intialState = {
  genreFilter: `all`,
  films
};

const ActionCreator = {
  changeGenre: (value) => {
    if (
      [
        `comedy`,
        `crime`,
        `documentary`,
        `drama`,
        `horror`,
        `family`,
        `romance`,
        `sci-fi`,
        `thriller`
      ].find((item) => item === value)
    ) {
      const filmsNew = films.filter((item) => item.genre === value);
      return {
        type: `CHANGE_GENRE`,
        genreNew: value,
        filmsNew
      };
    }
    return {
      type: `RESET`
    };
  }
};

const reducer = (state = intialState, action) => {
  switch (action.type) {
    case `CHANGE_GENRE`:
      return Object.assign({}, state, {
        genreFilter: action.genreNew,
        films: action.filmsNew
      });
    case `RESET`:
      return Object.assign({}, intialState);
  }
  return state;
};

export {ActionCreator, reducer};
