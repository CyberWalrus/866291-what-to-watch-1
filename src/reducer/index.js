import {combineReducers} from "redux";
import {reducer as filter} from "./filter/filter.js";
import {reducer as data} from "./data/data.js";
import NameSpace from "./name-spaces.js";

export default combineReducers({
  [NameSpace.FILTER]: filter,
  [NameSpace.DATA]: data
});
