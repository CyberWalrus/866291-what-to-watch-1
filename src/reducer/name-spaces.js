const NameSpace = {
  DATA: `DATA`,
  FILTER: `FILTER`
};

export default NameSpace;
