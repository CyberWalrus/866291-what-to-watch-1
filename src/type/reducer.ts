import {ThunkDispatch as ReduxThunkDispatch, ThunkAction as ReduxThunkAction} from "redux-thunk";
import {AxiosInstance} from "axios";
import {State as StateData, Action as ActionData} from "../store/data/data";
import {State as StateFilter, Action as ActionFilter} from "../store/filter/filter";
import {State as StateUser, Action as ActionUser} from "../store/user/user";
import NameSpace from "../store/name-spaces";

export interface StateApp {
  [NameSpace.DATA]: StateData;
  [NameSpace.FILTER]: StateFilter;
  [NameSpace.USER]: StateUser;
}

export type ActionApp = ActionData | ActionFilter | ActionUser;
export type ThunkDispatch = ReduxThunkDispatch<StateApp, AxiosInstance, ActionApp>;
export type ThunkAction = ReduxThunkAction<
Promise<void>,
StateApp,
AxiosInstance,
ActionApp
>;
