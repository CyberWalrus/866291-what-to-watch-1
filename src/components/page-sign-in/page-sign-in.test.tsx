import * as React from "react";
import * as Enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";
import {createStore} from "redux";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import PageSignIn from "./page-sign-in";
import reducer from "../../store";
Enzyme.configure({adapter: new Adapter()});

it(`PageSignIn correctly renders after relaunch`, (): void => {
  const store = createStore(reducer);
  const handleClick = jest.fn();
  const tree = toJson(
    shallow(
      <Provider store={store}>
        <BrowserRouter>
          <PageSignIn
            email={``}
            password={``}
            onChangeUserInput={handleClick}
            onClickSubmit={handleClick}
            formErrors={{}}
            formValid={false}
          />
        </BrowserRouter>
      </Provider>
    )
  );
  expect(tree).toMatchSnapshot();
});
