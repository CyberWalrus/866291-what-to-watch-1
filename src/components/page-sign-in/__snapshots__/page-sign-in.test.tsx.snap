// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSignIn correctly renders after relaunch 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <BrowserRouter>
    <PageSignIn
      email=""
      formErrors={Object {}}
      formValid={false}
      onChangeUserInput={[MockFunction]}
      onClickSubmit={[MockFunction]}
      password=""
    />
  </BrowserRouter>
</ContextProvider>
`;
